### PR TITLE
Some improvements on Migrations

### PR DESCRIPTION
**Pull request by ProGM**
This pull introduces/changes:
 * `add_index` and `add_constraint` helpers
 * A `populate_id_property` helper, as a replacement for the legacy `AddIdProperty` task
 * Splits migration helpers in submodules

Pings:
@cheerfulstoic
@subvertallchris

### CODE REVIEW

File: lib/neo4j/migration.rb [hunk 1]
```diff
@@ -23,24 +23,30 @@ def joined_path(path)
...
+      include Neo4j::Migrations::Helpers::IdProperty
+
       attr_reader :models_filename
 
       def initialize(path = default_path)
         @models_filename = File.join(joined_path(path), 'add_id_property.yml')
       end
 
       def migrate
+        ActiveSupport::Deprecation.warn '`AddIdProperty` task is deprecated and may be removed from future releases. '\
```
Comment (cheerfulstoic) [inline lib/neo4j/migration.rb]: I was actually thinking about putting the deprecation into 7.0.x and then removing it in 8.0.  What do you think?

**Comment by cheerfulstoic** [comment]
LGTM!  @subvertallchris ?

File: lib/neo4j/migration.rb (hunk 1 above)
Comment (ProGM) [inline lib/neo4j/migration.rb]: Yep, I though the same. I added the deprecation warning here, then we can remove the entire (old) `Neo4j::Migration` namespace from 8.0+ , if you agree :)

**Pull request merged**